NetKet/Graph/abstract_graph.hpp

```diff
@@ -49,11 +67,17 @@ class AbstractGraph {
   */
   virtual std::vector<std::vector<int>> SymmetryTable() const = 0;
 
+  /**
+  Custom type for unordered_map<array<int,2>, int> w/ a custom hash function
+  */
+  using ColorMap =
+      std::unordered_map<std::array<int, 2>, int, std::ArrayHasher>;
```

Comment:
Maybe even add another alias for the edge? (Then the reader can be sure immediately what the `std::array<int, 2>` is for.)
```cpp
using Edge = std::array<int, 2>;
using ColorMap = std::unordered_map<Edge, int, std::ArrayHasher>;
```